### DISCUSSION TITLE
Home page designs

### DISCUSSION THREAD

**Discussion by advplyr** [advplyr/audiobookshelf#63] [Ideas]
First attempts at a home page design. The difficulty is finding a label for the shelves that fits the bookshelf design.

These are just some sample categories, I think the shelves could be adjustable in the server settings.

I'm not set on any of these btw, I could really use some help on design so I can focus more on functionality. Thank you to all helping out.

![SampleHome6](https://user-images.githubusercontent.com/67830747/135010288-92f063d4-ce55-4a2f-98ce-80d01d9cebb3.png)

![SampleHome](https://user-images.githubusercontent.com/67830747/135005743-ba591fa9-b5c6-43ef-a4df-8cd488394fc2.png)

![SampleHome2](https://user-images.githubusercontent.com/67830747/135005746-e91d30fb-b9c9-4e2b-b535-c09fe536278b.png)

![SampleHome3](https://user-images.githubusercontent.com/67830747/135005753-08ec49d5-8812-4588-b053-a23f51042671.png)

**Comment by Jdiesel87** [comment]
I think my preference would be for the second one. It reminds me the most of something like this

![643598f876d46183a1dad15b366d2c01](https://user-images.githubusercontent.com/14824476/135008069-f0f12654-0307-43ba-90e2-8a4f3692d9e8.jpg)

If you increase the thickness of the shelf you might able to include the label on the shelf itself.

**Reply by Jdiesel87** [reply]
Maybe a personal preference but I like the inverse of my example. Black text on white background, it reminds me of a standard label from a label maker.

**Reply by advplyr** [reply]
Something like this perhaps?

![image](https://user-images.githubusercontent.com/67830747/135008911-68a8b19e-6baa-4e0e-a459-40e659226554.png)

**Reply by Jdiesel87** [reply]
Yeah that is kind of what I had in mind. Is that a static image or are those css elements?

**Reply by advplyr** [reply]
Or pushed left. These are CSS

**Reply by advplyr** [reply]
I can never seem to get light colors looking good, but could just be my own bias

![image](https://user-images.githubusercontent.com/67830747/135009556-315fdf57-8c34-4950-9a64-8ba9af15b221.png)

**Reply by Jdiesel87** [reply]
Tough call maybe split the difference. Not flush with the left edge of the first book but not flush with the left edge of the view.

**Reply by Jdiesel87** [reply]
Yeah placement just like that

**Reply by advplyr** [reply]
I think you're right. It is more realistic. I don't like how thick the shelf is, it makes the single color stand out more.
As of now I'm leaning towards this one, but will give it a night to think on it.

update: putting these images in the initial post

**Reply by Jdiesel87** [reply]
I think your are right with the black label. There are lots of warm colors so it makes the white tough to get right. Pure white is too jarring and getting the right amount of yellow to warm it up is a fine balance.

Besides I'm sure someone will end up doing some custom CSS to make it their own anyways.

**Reply by advplyr** [reply]
True, and I hope to have server customizations and themes built-in down the road.

**Reply by Budlyte** [reply]
Oh good, We agree on the shelf label being the winner! Even though I think the little hanging sign is really cute.  This initial run at the Home Screen looks good.  I want to suggest a conditional On Deck row under the Continue Reading row, that contains the next unread in a series that has at least one read, not all are read, and there are no in progress.

I think you guys were starting to get at what I believe is wrong with the shelf labels, the border washes out the ability to clearly read the text.  Have you tried dropping the border and doing a transparent background with high contrasting text against the wood color? 

Testing them [here](https://coolors.co/contrast-checker/ffffff-95775a), white & black seem to work about equally. That is to say nothing really contrasts well with that honey wood color.  LOL  It's such a damn neutral color, that this is going to be a tough one.  Adding something that needs to stand out on a theme that by its nature is very neutral.... well I'm not sure there is a good answer to this one.

**Reply by Budlyte** [reply]
Or, you could drop the wood theme, and add some color pickers like that website has in a UI section in Settings, letting the User do the work to set the colors for each segment:
- Toolbars
- Background
- Borders
- Text
- Highlighted Text

Just a thought, to allow you to be able to say "Here, pick whatever colors you people want and leave me alone.".   :)
Hmmm, rolling through the app components these items are already broken out.... Appbar, Bookshelf, Bookshelf Toolbar, Siderail.... Oh my.   <wink wink> 

I just wish I knew how to compile a darn program and get it running myself. Something like this change seems like a straight forward ordeal: Add new components for the color wheels, setting selections into variables, and then reference those variables in the style portion of the other objects.  I want to help but I lack the skills!  D:

**Reply by advplyr** [reply]
I think you are right about the border color being too bright. Looks better like this:

![image](https://user-images.githubusercontent.com/67830747/135075304-987fa10a-228c-4a06-b0c7-839d68b32de4.png)

![image](https://user-images.githubusercontent.com/67830747/135077178-e1f906a0-3bd5-46ad-ab7b-bdca7b92c5c2.png)

I also agree on allowing customization, but I think there needs to be a nice default first. A solid color book shelf divider looks bad too, it would need to support gradient customization. Customization needs to be persisted as well, so it needs to be stored in server settings and loaded.

**Reply by advplyr** [reply]
I pushed an update `v1.2.6`, but I didn't create a release for it because it isn't complete yet. This way you can look at it if you are around today, because I won't be able to complete it until this evening.

Some things incomplete:

- There is no ability to scroll if the books go off the page.
- Using the back arrow when on an audiobook page always goes to the library.
- Shelf categories were just examples

**Reply by Budlyte** [reply]
Yeah, that's a bit more clear. Looks like you made the font bigger.

The Continue Reading shelf displays books that aren't started and one that's finished.
![image](https://user-images.githubusercontent.com/4451365/135083662-3bf1ab45-0fc0-46eb-9caa-71881260054c.png)

I like the addition of the Newest Series shelf. I see that one being useful. And it's sorted by Series # too!

**Reply by advplyr** [reply]
Fixed the continue reading shelf in the last update.

**Comment by Jdiesel87** [comment]
Looks good!

Have you put any thought into how many books you will limit to each category? I haven't tried the recent update yet but I am wondering if you are using horizontal grid scrolling?

**Reply by advplyr** [reply]
Not much thought. I think I have it capped at 10 right now. It is set-up for scrolling but the scrollbar is hidden. I tried the scrollbar multiple ways and it just looks bad, so I think the way to go is to add an arrow on the right side that shows up when you hover near it.

**Reply by advplyr** [reply]
I went with the arrow strategy for scrolling, a bit of a pain to implement, we'll see how it goes.

**Comment by Jdiesel87** [comment]
I didn't see a commit for it but the new series view is much nicer with the carousel image thumbnails and collection titles when hovering

**Comment by jarrodCoombes** [comment]
I like the way it looks now, but #2 looks good too. I would love to replace the "Newest Series" shelf with one that is called "Updated Series" and it does not show individual books, rather shows the series tile itself (So Updated Series: "Wizards Rule" tile, you click on it and it takes you to the series).

I am trying to think about what could be useful on the homepage past what you already have, and I can't think of anything, but being able to create custom "shelves" to show up there based on certain criteria (maybe tags and/or genres) could be handy.

**Reply by advplyr** [reply]
Agree with the series shelf.

For those with multiple users there could be a shelf that shows what other users are reading. Not sure what the label could be, "Other users reading", doesn't work well.

My initial thought was it would be sort of like netflix. Where you can keep scrolling down and there is a shelf for each genre. Shelves for tags and maybe even authors. Also, the shelves would be configurable, so if you didn't want to show genre shelves you could disable them.
What do you think? Do you prefer just a few shelves or more like a netflix thing?

**Reply by Jdiesel87** [reply]
I would like to see user generated shelves using the existing filters and sorting. Limited it to however many shelves is appropriate, maybe 10-15, and let the user decide how to display them.

Custom Row 1
Genre - Autobiography     Sort - Added At

Custom Row 2
Genre - Thriller                  Sort - Added At

Custom Row 3
Author -  Stephen King     Sort - Title

Custom Row 4
Series - The Dark Tower    Sort - Volume #

Custom Row 5
Blank                                 Blank

Custom Row 6
Blank                                 Blank

and so forth. You can create pretty much any view you want from the existing filters. The only thing missing is maybe a "random" view or generator that is dynamic in the sense it changes every time you refresh the view.

**Reply by advplyr** [reply]
Great idea, build your own shelves type of thing. Do you think the server owner should be setting shelves or is this a per-user setting?
I'm also assuming this should be able to be configured differently per library. So a per-user and per-library config.

**Reply by Jdiesel87** [reply]
I think per user

**Reply by jarrodCoombes** [reply]
This could simply be the ability to add a shelf with any of the database fields. So you could have a "Stephen King" shelf (author), or a "Harry Potter" shelf (series), or a "Horror" shelf (genre), or a "Happy Books" shelf (tag) etc. And I agree capping it at 10 or so would be good. 

I do like the idea of a "What others are reading" shelf too.

**Reply by jarrodCoombes** [reply]
> Do you think the server owner should be setting shelves or is this a per-user setting?

It would be nice to have both options. The users can create their own, but the server owner (root user) has the ability to do create system wide shelves (may be limited to just 2, where as the user can create 10 or so, with the owners shelves always being on top).
